### DISCUSSION TITLE
Validated workflow doesnt work

### DISCUSSION THREAD

**Discussion by martinzerty** [projectdiscovery/nuclei#3893] [Q&A] [answered]
### Nuclei version:
v2.9.7

### Current Behavior:
I want to run a workflow with two templates. Each template works well separatly (except for warnings).
`nuclei -validate`
says everything is fine, even with the workflow. But when I want to launch the workflow, it says `[FTL] Could not run nuclei: no valid templates were found`

### Expected Behavior:
I wanted to use a workflow with 2 templates and a matcher.

### Steps To Reproduce:

login.yaml :
```
  id: hm-login-one
  
  info:
      name: send login form
      author: me
      severity: medium
      description: send login form
    
  
  requests:
    - raw:
      - |
        POST {{BaseURL}}?siteLanguage=fr HTTP/1.1
        Host: {{Hostname}}
        Accept-Language: fr,fr-FR;q=0.8,en-US;q=0.5,en;q=0.3
        Accept-Encoding: gzip, deflate
        Content-Type: application/x-www-form-urlencoded
        Connection: keep-alive
        Cookie: {{cookie}}
```

cookie-extracor.yaml:

```

id: cookie-detect
info:
  name: Cookies-extractor
  author: Nagli
  description: Extract cookies
  severity: info
  tags: cookies

requests:
  - method: GET
    path:
      - "{{BaseURL}}"

    extractors:
      - type: kval
        name: cookie
        part: header
        kval:
          - Set_Cookie

```

login_with_cookie.yaml

```
id: login-from-cookie
info:
  name: Login from Cookie
  author: martin

workflows:
  - template: cookie-extractor.yaml
 

```
- `nuclei -t login_with_cookie.yaml -validate`
gives : 
```
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.9.7

        projectdiscovery.io

[INF] All templates validated successfully
```

 - `nuclei -u [my url) -t login_with_cookie.yaml`
 gives : 
```
[INF] Current nuclei version: v2.9.7 (latest)
[INF] Current nuclei-templates version: v9.5.3 (latest)
[INF] Targets loaded for current scan: 1
[INF] No results found. Better luck next time!
[FTL] Could not run nuclei: no valid templates were found
```

Can anyone help ?

**Comment by MetzinAround** [comment]
This is a long shot, but the name of the second file is `cookie-extracor.yaml:`

Is it possible it's misspelled on the save file as well, or is that just a typo in this issue?

**Reply by martinzerty** [reply]
I made multiple changes. It might have helped. Thank you !

**Comment by ehsandeep** [comment, marked as answer]
@martinzerty This is unrelated to validate option.

1. Templates and workflows are different types, you need to use `-w` option to run/validate workflows instead of `-t`
2. You can use `-v` option to see more information about errors.

**Reply by martinzerty** [reply]
Thanks a lot !
